app/routes/url.tsx

```diff
@@ -28,17 +28,33 @@ enum Action {
   DECODE = "Decode",
 }
 
+function textToUrl(text: string) {
+  return text.startsWith("https%3A%2F%2F") ? decodeURIComponent(text) : text;
```

Comment:
Haven't merged this yet because there is likely a way better check. This only works for https.